app.rb

```diff
@@ -0,0 +1,192 @@
+require_relative './modules/rental'
+require_relative './book'
+require_relative './person'
+require_relative './modules/student'
+require_relative './modules/classroom'
+require_relative './modules/teacher'
+
+class App
+  attr_reader :people, :books, :rentals
+
+  def initialize()
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "Title: #{book.title},  Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      case person
+      when Student
+        puts "[Student] Name: #{person.name}, ID: #{person.id}, Age: #{person.age},
+         Parent Person: #{person.parent_permission}"
+      when Teacher
+        puts "[Teacher] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person(who, age, name, parent_permission, specialization = nil)
+    case who
+    when 'Student'
+      person = Student.new(age, name, parent_permission)
+      puts '----------------------------------'
+      puts "Student:  Name:  #{person.name}"
+      puts "          Age:  #{person.age}"
+      puts "          Parent Permission: #{person.parent_permission}"
+      puts 'Student created succesfully'
+    when 'Teacher'
+      person = Teacher.new(name, age, specialization, parent_permission)
+      puts '----------------------------------'
+      puts "Teacher:  Name:  #{person.name}"
+      puts "          Age:  #{person.age}"
+      puts "          Specialization:  #{person.specialization}"
+      puts "          Parent Permission:  #{person.parent_permission}"
+      puts 'Teacher created succesfully'
+    end
+    @people << person
+  end
+
+  def create_book(title, author)
+    puts '----------------------------------'
+    book = Book.new(title, author)
+    puts "Book: Title: #{book.title}"
+    puts "      Author: #{book.author}"
+    @books << book
+    puts 'Book created succesfully'
+  end
+
+  def create_rental(book_index, person_index, d_date)
+    book = @books[book_index]
+    person = @people[person_index]
+    date = d_date
+    @rentals << Rental.new(date, book, person)
+    puts 'Rental Created.'
+  end
+
+  def list_rentals(_id)
```

Comment:
- [ ] Great work on the app, however, the app breaks when trying to list the rentals, kindly make sure to fix this so the user can list the available rentals

- the error 👇🏼 
![Screenshot_20230320_194039](https://user-images.githubusercontent.com/25227022/226447860-790e2fca-c9f9-452e-b439-588145dec08b.png)